core/issuerservice/issuerservice-credentials/src/main/java/org/eclipse/edc/issuerservice/credentials/statuslist/bitstring/BitstringStatusListManager.java

```diff
@@ -179,16 +177,16 @@ private VerifiableCredentialResource createCredentialResource(String participant
     }
 
     private Result<VerifiableCredentialResource> publish(VerifiableCredentialResource statusListCredential) {
-        var participantContextId = statusListCredential.getParticipantContextId();
-        var urlResult = publisher.publish(participantContextId, statusListCredential.getId());
-        return urlResult.compose(url -> {
-            var meta = new HashMap<>(statusListCredential.getMetadata());
-            meta.put(PUBLIC_URL, url);
-            var updated = statusListCredential.toBuilder().metadata(meta).build();
-            var storeResult = upsert(updated);
-            return storeResult.succeeded() ?
-                    Result.success(updated) : Result.failure(storeResult.getFailureDetail());
-        });
+        return publisher.publish(statusListCredential)
+                .compose(url -> {
+                    var meta = new HashMap<>(statusListCredential.getMetadata());
+                    meta.put("published", true);
```

Comment:
I'd prefer to set the `"published"` entry in the metadata map directly in the `LocalCredentialPublisher`

Reply:
I think we should try to isolate changes to the metadata in a single place, and the looks to be the best place to do, doing that in the publisher would be a sort of unexpected side effect, the publisher is just expected to publish, then if the publish goes through, the manager will update metadata, status and what's needed.

plus, if we're considering the possibility of different publisher implementations, doing that in the caller makes even more sense.

Reply:
similarly, we'd have to re-implement this entire behavior for status list implementations other than BitStringStatusList, e.g. StatusList2021, which is definitely planned.

I'd say the chances of ever implementing an addition publisher are magnitudes lower than additional status list impls. By that logic, I'd probably even move both the public url and the "published" flag into the publisher, because that were it semantically belongs.

Reply:
this is a design issue, in fact the way the different status list kinds are managed should be the same and only the implementation specific behavior should finish into a specification, so describing it following the  abstraction pattern (that's not the best way to go but the easy to explain the concept in this moment) there would be an abstract `StatusListManager` that contains this common logic that:
- calls the provider
- updates the metadata

then the two extensions `BitStringStatusListManager` and `StatusList2021Manager` will contain the related custom logic.

Reply:
yep, that sounds reasonable. such a refactoring could be done in a subsequent PR, or in this, either way is fine with me.

Reply:
It could be done when the `StatusList2021` implementation will be done, I wouldn't put additional pressure on this bugfix PR